Issue with running `docker run` for integration testing

I have run into some issue with running `docker run xx-image` after moving some of the workflow into self-hosted runner. I could not figure it out, but feel it might be helpful to post in here and discuss.

The flow looks like something below:
```
docker run -it --rm \
	-v $(PWD):/local \
	xx-image:latest \
	/local/e2e.sh
```

The error log that i keep getting:

```
docker: Error response from daemon: OCI runtime create failed: container_linux.go:349: starting container process caused "exec: \"/local/e2e.sh\": stat /local/e2e.sh: no such file or directory": unknown.
```

While it works before, I suspect it might be related to the `dind` docker image it used for the runner. Does the dind docker engine impact the volume mount?? 

Please shed some insight on fixing the issue and let me know if this makes sense. Thanks!!

I dont think I have `/github/workflow` path there.

Here is the debug output:
```
Run find -maxdepth 3 /github/workspace
  find -maxdepth 3 /github/workspace
  shell: /bin/bash -e {0}
find: paths must precede expression: `/github/workspace'
##[error]Process completed with exit code 1.
```

In my test PR, I am still seeing the same thing

```
Run find -maxdepth 3 /github/workspace
  find -maxdepth 3 /github/workspace
  shell: /bin/bash -e {0}
find: paths must precede expression: `/github/workspace'
##[error]Process completed with exit code 1.
```

--- a/controllers/runner_controller.go
+++ b/controllers/runner_controller.go
@@ -282,6 +282,10 @@ func (r *RunnerReconciler) newPod(runner v1alpha1.Runner) (corev1.Pod, error) {
 					EnvFrom:         runner.Spec.EnvFrom,
 					VolumeMounts: []corev1.VolumeMount{
 						{
+							Name:      "work",
+							MountPath: "/runner/_work",
+						},
+						{
 							Name:      "docker",
 							MountPath: "/var/run",
 						},
@@ -296,6 +300,10 @@ func (r *RunnerReconciler) newPod(runner v1alpha1.Runner) (corev1.Pod, error) {
 					Image: r.DockerImage,
 					VolumeMounts: []corev1.VolumeMount{
 						{
+							Name:      "work",
+							MountPath: "/runner/_work",
+						},
+						{
 							Name:      "docker",
 							MountPath: "/var/run",
 						},
@@ -306,7 +314,13 @@ func (r *RunnerReconciler) newPod(runner v1alpha1.Runner) (corev1.Pod, error) {
 				},
 			},
 			Volumes: []corev1.Volume{
-				corev1.Volume{
+				{
+					Name: "work",
+					VolumeSource: corev1.VolumeSource{
+						EmptyDir: &corev1.EmptyDirVolumeSource{},
+					},
+				},
+				{
 					Name: "docker",
 					VolumeSource: corev1.VolumeSource{
 						EmptyDir: &corev1.EmptyDirVolumeSource{},

